go/app/main.go

```diff
@@ -20,22 +24,116 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type Item struct {
+	ID 	     int    `json:"id"`
+	Name     string `json:"name"`
+	Category string `json:"category"`
+	Image    string `json:"image_name"`
+}
+
+type Items []Item
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
 func addItem(c echo.Context) error {
 	// Get form data
+    id := c.FormValue("id")
+    c.Logger().Infof("Receive id: %s", id)
+
 	name := c.FormValue("name")
 	c.Logger().Infof("Receive item: %s", name)
 
-	message := fmt.Sprintf("item received: %s", name)
+	category := c.FormValue("category")
+	c.Logger().Infof("Receive category: %s", category)
+
+	image, err := c.FormFile("image")
+	if err != nil {
+		c.Logger().Error("Failed to receive image file")
+	}
+	c.Logger().Infof("Receive image: %s", image)
+
+	// Create a SHA256 hash
+	hash := sha256.New()
+
+	src, err := image.Open()
```

Comment:
Is this uploaded file saved? It's a temporary file and it will be lost unless you save it somewhere and you can't fulfill this requirement:

> ポストされた画像のファイルを sha256 で hash化し、<hash>.jpgという名前で保存します

Reply:
Sorry, I missed that part! How about this implementation?

https://github.com/F-0728/mercari-build-training/blob/3cdefb398ed4c82789ca94045a75340d3cae2953/go/app/main.go#L74-L79

Reply:
You're so close, but won't it create an empty file?

Reply:
Seems I misunderstood the requirement.
Was I supposed to hash the filename only, and the content should remain?
Just like changing the filename of the image?

Reply:
Ah, in that sense what's written in the requirement can be misleading...
But the content needs to be saved into the disk! Having a empty file (size == 0) like this benefits no one.

```bash
% ls -l images
total 16
-rw-r--r--@ 1 mshibuya  staff     0  2 20 12:20 ad55d25f2c10c56522147b214aeed7ad13319808d7ce999787ac8c239b24f71d.jpg
-rw-r--r--@ 1 mshibuya  staff  5597  2 17 15:23 default.jpg
```

Reply:
Yes, indeed. I should have questioned on that point.
Now the image must be copied!

https://github.com/F-0728/mercari-build-training/blob/de0bde4bacbffe2e99f2d98356a13abaeae407cd/go/app/main.go#L59-L84